### ISSUE TITLE
Advanced atom query features don't roundtrip through SDF

### ISSUE THREAD

**Issue by rachelnwalker** [rdkit/rdkit#5718]
**Describe the bug**
RDKit's SD parsing allows for smarts atom queries to be represented using substance groups (like in [this test](https://github.com/rdkit/rdkit/blob/550411c49358f787fda60d081bd8fbf16632e3c0/Code/GraphMol/FileParsers/file_parsers_catch.cpp#L4040), looks like the actual parsing is done [here](https://github.com/rdkit/rdkit/blob/42eed2605dbd5fb6ce03bd72ab714c8a537dfea5/Code/GraphMol/FileParsers/MolFileParser.cpp#L2768).) However, when writing back to SDF, the smarts queries are not translated back to a substance group so the information is lost.

**To Reproduce**
```
from rdkit import Chem
mb = """

  Mrv2108 07152116012D          
  0  0  0     0  0            999 V3000
M  V30 BEGIN CTAB
M  V30 COUNTS 2 1 1 0 0
M  V30 BEGIN ATOM
M  V30 1 C -0.8333 4.5421 0 0
M  V30 2 C 0.5003 5.3121 0 0
M  V30 END ATOM
M  V30 BEGIN BOND
M  V30 1 1 1 2
M  V30 END BOND
M  V30 BEGIN SGROUP
M  V30 1 DAT 0 ATOMS=(1 2) -
M  V30 FIELDDISP="    0.0000    0.0000    DR    ALL  0       0" -
M  V30 QUERYTYPE=SMARTSQ QUERYOP== FIELDDATA=[#6;R]
M  V30 END SGROUP
M  V30 END CTAB
M  END"""
m = Chem.MolFromMolBlock(mb)

# the smarts query feature works!
assert Chem.MolFromSmiles("C1CCCCCC1").HasSubstructMatch(m)
assert not Chem.MolFromSmiles("CCC").HasSubstructMatch(m)

# the substance group is not present, so if we read this file back in,
# it would not have the smarts query
output_mb = Chem.MolToMolBlock(m,forceV3000=True)
print(output_mb)
```
output:

```
     RDKit          2D

  0  0  0  0  0  0  0  0  0  0999 V3000
M  V30 BEGIN CTAB
M  V30 COUNTS 2 1 0 0 0
M  V30 BEGIN ATOM
M  V30 1 C -0.833300 4.542100 0.000000 0
M  V30 2 C 0.500300 5.312100 0.000000 0
M  V30 END ATOM
M  V30 BEGIN BOND
M  V30 1 1 1 2
M  V30 END BOND
M  V30 END CTAB
M  END
```

**Expected behavior**
The output SDF should include the substance group with the atom query smarts.

**Additional context**
I feel like it would be nice to always include this information in outputted SDFs, but I could also see adding an option to the writer to include query information if there is a reason we wouldn't always want it.

**Comment by greglandrum** [comment]
Agreed that there should be some way to get those properties into the output SDF.
I think it's worth discussing if there should be an option for this or if we should just always do it. I'm leaning towards the latter, though that would definitely be a break with current behavior.
@ptosco, @bp-kelley : any thoughts here?

**Comment by bp-kelley** [comment]
I think an option would be great and also think it should default to being on.  I like the idea of least surprise so if we parse it and handle it, we should output it.

**Comment by greglandrum** [comment]
Since we can't add the option without changing the API, and this would mean this fix has to wait until the 2023.03 release, I'm going to do this in two stages:
1. start writing the query information to the CTAB
2. create a separate PR which adds the option.

**Comment by greglandrum** [comment]
Ok, what I ended up doing in #5825 is keeping track of when `SMARTSQ`s have been read and only writing that information back out to the Mol block. This shouldn't result in changes in the output for most people and still allows this information to be read and written.
#5819 is the enhancement request to come up with a generic mechanism for writing atom SMARTS to SMARTSQ blocks.
